scipy/linalg/matfuncs.py

```diff
@@ -668,3 +675,329 @@ def rounded_sign(x):
         return S0
     else:
         return S0, errest
+
+
+def is_matrix_hermitian(A):
+    """
+    Returns True if a matrix is hermitian
+    Parameters
+    ----------
+    A : (N, N) array_like
+        Matrix to evaluate.
+    
+    Returns
+    -------
+    X : boolean
+        True if a matrix A is hermitian.
+
+    Examples
+    --------
+    >>> from scipy.linalg import is_matrix_hermitian
+    >>> m = np.array([[0,1j],[-1j,0]])
+    >>> print(is_matrix_hermitian(m))
+    True
+    >>> m = np.array([[5,0],[1,3]])
+    >>> print(is_matrix_hermitian(m))
+    False
+    """
+    A = _asarray_square(A)
+    return np.sum(A == A.conj().T) == (A.shape[1]**2)
```

Comment:
This can only work if it is numerically close. Besides it is a one liner so I am not sure if we even need this. You can check this via `np.any()` by the way.

Reply:
Hi @ilayn, thank you for the feedback! Please consider this as the first sprint.
The new version of "is_matrix_positive_definite" has improvements that could be useful, such as, it allows the user to specify the algorithm for testing definitness.
If this is a good design then it can be applied to some of the other methods (the most useful ones).